File: LeetCode_Solutions/LeetCode_Solutions/ContainsDuplicate_217.m
//
//  ContainsDuplicate_217.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/10/25.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//


#import "ContainsDuplicate_217.h"

@implementation ContainsDuplicate_217

- (BOOL)containsDuplicate:(NSArray *)array {

    NSInteger count = 0;

    for (int i = 0; i < array.count; i++) {
        for (int j = i + 1; j < array.count; j++) {
            if ([array[i] integerValue] == [array[j] integerValue]) {
                count ++;
            }
        }
    }
    
    if (count >= 1) {
        return YES;
    }
    return NO;
}

@end

File: LeetCode_Swift/LeetCode_Swift/ContainsDuplicate_217.swift
//
//  ContainsDuplicate_217.swift
//  LeetCode_Swift
//
//  Created by Huni on 15/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//


/*
 
    这里可以利用Set的特性：集合中没有重复的元素，这一个特点来判断数组中是否有重复元素
 
 */

import Foundation

public class ContainsDuplicate_217 {

    func containsDuplicate(_ nums: [Int]) -> Bool {
        
        return nums.count > Set(nums).count
    }
}

